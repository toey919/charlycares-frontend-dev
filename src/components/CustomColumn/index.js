// @flow

import React from 'react';
import styled from 'styled-components';
import { Grid } from 'semantic-ui-react';

type Props = {
  noPadding: boolean,
  padding: string,
  fullStretch: boolean,
};

export const CColumn: Grid.Column = ({
  noPadding,
  padding,
  fullStretch,
  ...rest
}: {
  noPadding: boolean,
  padding: string,
  fullStretch: boolean,
  rest: Array<Object>,
}) => <Grid.Column {...rest} />;

const CustomColumn = styled(CColumn)`
  &&& {
    padding: ${(props: Props): string | number =>
      props.noPadding ? 0 : props.padding && props.padding} !important;
    margin-left: ${props => props.fullStretch && '-1rem'};
    margin-right: ${props => props.fullStretch && '-1rem'};
    margin-bottom: ${props => (props.marginbottom ? props.marginbottom : 'none')};
    position: relative;
    text-align: ${props => (props.textAlign ? props.textAlign : 'none')};
  }
`;

export default CustomColumn;
