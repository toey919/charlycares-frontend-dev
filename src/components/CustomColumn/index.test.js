import React from 'react';
import { shallow } from 'enzyme';
import CustomColumn, { CColumn } from './index';

describe('CustomColumn', () => {
  it('Should render styled component', () => {
    let wrapper = shallow(<CustomColumn />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Should render component without custom props', () => {
    let wrapper = shallow(<CColumn noPadding padding="20px" />);
    expect(wrapper).toMatchSnapshot();
  });
});
