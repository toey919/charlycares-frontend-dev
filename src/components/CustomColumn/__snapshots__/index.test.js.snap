// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomColumn Should render component without custom props 1`] = `<GridColumn />`;

exports[`CustomColumn Should render styled component 1`] = `
<CColumn
  className="sc-bdVaJa dLnFPs"
/>
`;
